agent/app/agent.go

```diff
@@ -282,19 +282,25 @@ func (agent *ecsAgent) newTaskEngine(containerChangeEventStream *eventstream.Eve
 
 	// previousState is used to verify that our current runtime configuration is
 	// compatible with our past configuration as reflected by our state-file
-	previousState, err := agent.newStateManager(previousTaskEngine, &previousCluster,
+	previousStateManager, err := agent.newStateManager(previousTaskEngine, &previousCluster,
```

Comment:
* Comment needs to be updated alongside
* `previousStateManager` is confusing, a better name is preferable

Reply:
Good callout.

I'm not sure what is confusing about it; what would you recommend? I added 'manager' because its a manager object and not the underlying state.

Reply:
I was slightly confused; I'm no longer unhappy with the name
The `previousStateManager` uses the `previousTaskEngine` to become the new `taskEngine`

Reply:
Yeah. Its kind of weird but the whole function has this notion of 'previous' where everything labeled 'previous' is inferred from the state file that has been loaded